### PR TITLE
Unit tests for Draw and Deal cards

### PR DESCRIPTION
**Pull request by bjsvedin**
Added unit tests for draw and deal cards. Added a couple checks when drawing for empty draw pile and discard pile. Moved helper function to the bottom of the page. Added many comments.

### CODE REVIEW

**Comment by logan-hendricks** [comment]
@bjsvedin you will need to merge the latest dev. sorry. I just pulled in a Pull Request. However. If you merge that in now. I can likely get this out tonight.

File: server/deckUtils.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func generateShuffledDeck(numPlayers int) []model.Card {
 
 	for cardValue, count := range wildCardCounts {
 		for i := 0; i < count; i++ {
-			deck = append(deck, model.Card{"wild", cardValue})
+			deck = append(deck, model.Card{"black", cardValue})
```
Comment (logan-hendricks) [inline server/deckUtils.go:85]: I'm not certain if I understand. Shouldn't this be "wild". why would this be changed to "black"?

Comment (logan-hendricks) [reply server/deckUtils.go:85]: ah... I now understand... as this is the color

**Comment by bjsvedin** [comment]
I have merged in the latest from Dev.

**Comment by logan-hendricks** [comment]
> @logan-hendricks I have merged in the latest from Dev.

@bjsvedin solid. testing this now.

File: server/deckUtils.go (hunk 1 above)
Comment (bjsvedin) [reply server/deckUtils.go:85]: Yeah that was from a different item. I know the pull request is titled unit tests, but this branch is actual-uno-deck so there is a couple more items in it such as this.

Comment (bjsvedin) [reply server/deckUtils.go:85]: That will go along with the create new card guis once its merged. It originally was gonna merge into this branch but it was moved to dev.

Comment (logan-hendricks) [reply server/deckUtils.go:85]: that makes sense.

File: server/uno.go [hunk 2]
```diff
@@ -280,6 +240,47 @@ func dealCards(game *model.Game) (*model.Game, error) {
 	return game, err
 }
 
+func checkForCardInHand(card model.Card, hand []model.Card) bool {
```
Comment (logan-hendricks) [inline server/uno.go:243]: Did you want to mark this as a section for utility functions? or just have them at the bottom of the file. I'm really ok either way. I just wanted to see which way you wanted it

Comment (bjsvedin) [reply server/uno.go:243]: Lets mark it as utility functions

Comment (bjsvedin) [reply server/uno.go:243]: They were just scattered through the file and i wanted to group them together.

Comment (logan-hendricks) [reply server/uno.go:243]: k. let me know when you've pushed up that section heading.

Comment (bjsvedin) [reply server/uno.go:247]: Seaction header added to utility functions

**Pull request merged**